src/azure-cli/azure/cli/command_modules/rdbms/_params.py

```diff
@@ -465,5 +475,62 @@ def _flexible_server_params(command_group):
         with self.argument_context('{} flexible-server replica stop-replication'.format(command_group)) as c:
             c.argument('server_name', options_list=['--name', '-n'], help='Name of the replica server.')
 
+    def handle_migration_parameters(command_group, server_name_arg_type, subscription_arg_type, migration_id_arg_type):
+        for scope in ['create', 'show', 'list', 'update', 'delete']:
+            argument_context_string = '{} flexible-server migration {}'.format(command_group, scope)
+            with self.argument_context(argument_context_string) as c:
+                c.argument('subscription_id', arg_type=subscription_arg_type, options_list=['--subscription-id'],
```

Comment:
what about `--target-subscription` for this parameter as there is an existing parameter `--subscription` in CLI commands, which is the subscription where the resource you are operating in? And there is prefix match in azure cli, it will be confused with such two parameters starting from `--subscription`.

Reply:
Replaced it with the CLI param --subscription. Removed --subscription-id